### PR TITLE
Enhancement: Make UI/UX changes to `Blocked Reasons` filter.

### LINKED ISSUE #377: Enhancement: Add `Cookie Blocked` filter and make UI/UX changes to filters.
During the last implementation, the "Blocked" filter was removed and replaced with the "Cookie Blocked Reasons"
 filter. The idea was that users could use the second filter to access blocked cookies data. However, considering the use case, some users might want to see all blocked cookies regardless of the reason for blocking. In this scenario, the "Blocked" filter would be very beneficial.

Additionally, a few UI/UX changes are proposed for the Cookie Blocked Reasons filter. These changes include making it dynamic, alphabetically sorting the options, and renaming it from "Cookies Blocked Reasons" to "Blocked Reasons". Furthermore, the 'i' icon will be removed.

**Acceptance Criteria:**
- [x]  Introduce a static "Blocked" filter (value: True, False) to handle issue #335.
- [x]  Show only the blocked reasons that are actually used dynamically.
- [x]  Ensure that dynamic filters are sorted alphabetically.
- [x]  Rename "Cookies Blocked Reasons" to "Blocked Reasons" and remove the 'i' icon.

Related https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/371

### PR DESCRIPTION
**Pull request by mayan-000**
## Description
This PR makes the `Blocked Reasons` filter dynamic, sorts the dynamic filters, renames "Cookies Blocked Reasons" to "Blocked Reasons" and removes the 'i' icon.

## Screenshot/Screencast
Screenshot 2024-01-22 at 5 40 41 PM

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Fixes #377

### CODE REVIEW

File: packages/extension/src/view/devtools/components/cookies/cookiesListing/index.tsx [hunk 1]
```diff
@@ -36,7 +36,6 @@ import {
  * Internal dependencies.
  */
 import { useCookieStore } from '../../../stateProviders/syncCookieStore';
-import { BLOCKED_REASON_LIST } from '../../../../../constants';
```
Comment (ayushnirwal) [inline packages/extension/src/view/devtools/components/cookies/cookiesListing/index.tsx:39]: I see no other references to this constant in the project. Maybe we can delete it.

**Review by ayushnirwal — APPROVED** [review]
LGTM

**Comment by mohdsayed** [comment]
@mayan-000 Lets add some test coverage for it.

**Comment by mohdsayed** [comment]
JFI, In https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/381 we may add one filter "Blocked In" with "Request" and “Response”, then this "Blocked" filter wouldn't be required as you can view all blocked cookies by check both of these options.

**Review by mohdsayed — APPROVED** [review]

**Pull request merged**